web_external/js/views/body/MapPanel.js

```diff
@@ -115,6 +134,10 @@ minerva.views.MapPanel = minerva.View.extend({
             this.legendWidget[datasetId].remove(datasetId);
             delete this.legendWidget[datasetId];
         }
+        var layerIndex = $.inArray(layer, this.map.featureInfoWidget.layers);
```

Comment:
Thanks @mbertrand , let's keep it this way for now so we can get this in sooner.